### ISSUE TITLE
CoffeeScript preprocessor fails with moo lexer

### ISSUE THREAD

**Issue by loveencounterflow** [kach/nearley#412]

**Comment by loveencounterflow** [comment]
I suspect the issue to come from around https://github.com/kach/nearley/blob/202153f257fbe6583a9bc08ae0a887bbf93b89eb/lib/generate.js#L76

**Comment by tjvr** [comment]
This was provided for backwards-compatibility reasons. `%foo` can also be used for a custom token matcher, which is an object with a test() method, eg `var foo = {test: Tok -> Boolean}`.

Custom token matchers aren't generally recommended, and indeed Moo has deprecated the `has()` method (it now always returns `true`).

I'd happily accept a patch which updates generate.js 🙂

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by loveencounterflow** [comment]
Yeah that should definitely be patched. A `has()` method that always returns true is utterly misleading, and the conditional should be replaced by its `true` branch.

I couldn't find the spot where the ternary gets outputted, though. As a hot fix I'm piping the output of `nearleyc` through `sed --regexp-extended 's/\(lexer.has\("[^"]+"\) \? (\{type: "[^"]+"\}) : [^"]+\)/\1/g'`, but that can't be advertised as a solution, of course.
